.github/workflows/ci.yml

```diff
@@ -153,6 +153,53 @@ jobs:
 
     - run: make rust-benchmark
 
+  build_wasm_emscripten:
+    name: build wasm emscripten
+    # only run on push to main and on release
+    if: "success() && (startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/main')"
+    needs: [test, lint]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          components: rust-src
+          target: wasm32-unknown-emscripten
+          override: true
+
+      - uses: mymindstorm/setup-emsdk@v11
+        with:
+          version: 3.1.13
+          actions-cache-folder: emsdk-cache
+
+      - name: set up python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: set package version
+        run: python .github/set_version.py
+        if: "startsWith(github.ref, 'refs/tags/')"
+
+      - name: Sync Cargo.lock
+        run: cargo update -p pydantic-core
+        if: "startsWith(github.ref, 'refs/tags/')"
+
+      - run: pip install -U --pre maturin
+
+      - name: build wheels
+        run: maturin build --release --target wasm32-unknown-emscripten --out dist -i 3.10
```

Comment:
> It might be sensible to disable ABI3 for Emscripten wheels for now

I think that should be done in pyo3 if desirable. cc @davidhewitt 

And pydantic-core doesn't support ABI3 so it's not a problem in here.

Reply:
Of course it's harmless to generate abi3 wheels as long as the cost isn't very high.